api/net/Areas/Subscriber/Controllers/NotificationController.cs

```diff
@@ -0,0 +1,95 @@
+[ApiController]
+[Area("subscriber")]
+[ApiVersion("1.0")]
+[Route("api/v{version:apiVersion}/[area]/notifications")]
+[Route("api/[area]/notifications")]
+[Route("v{version:apiVersion}/[area]/notifications")]
+[Route("[area]/notifications")]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Unauthorized)]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
+public class NotificationController : ControllerBase
+{
+    #region Variables
+    private readonly IUserService _userService;
+    private readonly INotificationService _notificationService;
+    private readonly IKafkaMessenger _kafkaProducer;
+    private readonly KafkaOptions _kafkaOptions;
+    private readonly JsonSerializerOptions _serializerOptions;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a NotificationController object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="userService"></param>
+    /// <param name="notificationService"></param>
+    /// <param name="kafkaProducer"></param>
+    /// <param name="kafkaOptions"></param>
+    /// <param name="serializerOptions"></param>
+    public NotificationController(IUserService userService, INotificationService notificationService, IKafkaMessenger kafkaProducer, IOptions<KafkaOptions> kafkaOptions, IOptions<JsonSerializerOptions> serializerOptions)
+    {
+        _userService = userService;
+        _notificationService = notificationService;
+        _kafkaProducer = kafkaProducer;
+        _kafkaOptions = kafkaOptions.Value;
+        _serializerOptions = serializerOptions.Value;
+    }
+    #endregion
+
+    #region Endpoints
+    /// <summary>
+    /// Send the notification to the specified email address.
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="contentId"></param>
+    /// <param name="to"></param>
+    /// <returns></returns>
+    [HttpPost("{id}/send")]
+    [HttpPost("{id}/send/{contentId}")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(NotificationModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "Notification" })]
+    public async Task<IActionResult> SendToAsync(int id, long? contentId, string to)
+    {
+        var notification = _notificationService.FindById(id) ?? throw new NoContentException();
+
+        var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
+        var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
+
+        var request = new NotificationRequestModel(NotificationDestination.NotificationService, new { })
+        {
+            NotificationId = notification.Id,
+            ContentId = contentId,
+            RequestorId = user.Id,
+            To = to,
```

Comment:
`to` will be provided by the user returned by `userId`.

Reply:
I don't think that's the case, because it will be sent to a colleague, if I set it to the user object, it will be sent to it's own email.

Reply:
I'm pretty sure that Jeremy is referring to the new `userId` parameter he has suggested on the route.  On this topic though you did mention sharing to multiple colleagues during standup. So that might end up being an array of `colleague userId` ???

Reply:
The property **to** from the NotificationRequestModel requires a string with the email address. Since the emails will be sent individually, no arrays are required.
![image](https://github.com/bcgov/tno/assets/26801490/4d64095d-e01d-49e9-af6d-94b815f99eeb)
Looks like the notification object is correct.
I just want to double (triple) check it ...

Reply:
The `NotificationRequestModel` requires a string.
The endpoint you are building is to share a story with a colleague.  A colleague must be an existing user in the solution.  When a user wants to share a story, they must provide the `userId` of the user they want to share with.  You can then find the `email` for the colleague.

Reply:
Ok ! Got it !